cli/command/service/opts.go

```diff
@@ -579,6 +607,15 @@ func (options *serviceOptions) ToStopGracePeriod(flags *pflag.FlagSet) *time.Dur
 	return nil
 }
 
+// ToService takes the set of flags passed to the command and converts them
+// into a service spec.
+//
+// Takes an API client as the second argument in order to resolve network names
+// from the flags into network IDs.
+//
+// Returns an error if any flags are invalid or contradictory.
+//
+// nolint: gocyclo
```

Comment:
I think that if `gocyclo` is not happy here, that's a sign that we should refactor this function and split it into smaller pieces 😸

Reply:
Refactored this a hair. Having trouble getting `lint` to work right on my machine (as a consequence of having trouble getting `docker` working right on my machine, I'm using WSL2), so I'm using the CI system to check that my change works.